c:geo frequently crashing

User on support reports that c:geo crashes frequently for them. They seem to be able to do some steps, but on entering user note (?, Google translated) it's crashing again.

Their logfile has the following two types of error messages, each of them several times:

```
04-16 09:41:10.653  7557  7557 I zygote64: Rejecting re-init on previously-failed class java.lang.Class<Ld2>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/webkit/PacProcessor;
04-16 09:41:10.653  7557  7557 I zygote64:   at java.lang.Class java.lang.Class.classForName(java.lang.String, boolean, java.lang.ClassLoader) (Class.java:-2)
04-16 09:41:10.653  7557  7557 I zygote64:   at java.lang.Class java.lang.Class.forName(java.lang.String, boolean, java.lang.ClassLoader) (Class.java:453)
04-16 09:41:10.653  7557  7557 I zygote64:   at java.lang.Class android.webkit.WebViewFactory.getWebViewProviderClass(java.lang.ClassLoader) (WebViewFactory.java:150)
04-16 09:41:10.653  7557  7557 I zygote64:   at java.lang.Class android.webkit.WebViewFactory.getProviderClass() (WebViewFactory.java:417)
04-16 09:41:10.653  7557  7557 I zygote64:   at android.webkit.WebViewFactoryProvider android.webkit.WebViewFactory.getProvider() (WebViewFactory.java:211)
04-16 09:41:10.653  7557  7557 I zygote64:   at android.webkit.WebViewFactoryProvider android.webkit.WebView.getFactory() (WebView.java:2467)
04-16 09:41:10.653  7557  7557 I zygote64:   at java.lang.String android.webkit.WebView.findAddress(java.lang.String) (WebView.java:1738)
04-16 09:41:10.653  7557  7557 I zygote64:   at void android.text.util.Linkify.gatherMapLinks(java.util.ArrayList, android.text.Spannable) (Linkify.java:805)
04-16 09:41:10.653  7557  7557 I zygote64:   at boolean android.text.util.Linkify.addLinks(android.text.Spannable, int) (Linkify.java:481)
04-16 09:41:10.653  7557  7557 I zygote64:   at boolean android.text.util.Linkify.addLinks(android.widget.TextView, int) (Linkify.java:539)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.setPersonalNote(android.widget.TextView, android.view.View, java.lang.String) (CacheDetailActivity.java:2887)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.access$3300(android.widget.TextView, android.view.View, java.lang.String) (CacheDetailActivity.java:186)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.CacheDetailActivity$DescriptionViewCreator.setContent() (CacheDetailActivity.java:1699)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.activity.TabbedViewPagerFragment.notifyDataSetChanged() (TabbedViewPagerFragment.java:63)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.activity.TabbedViewPagerActivity.setOrderedPages(long[]) (TabbedViewPagerActivity.java:119)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.notifyDataSetChanged() (CacheDetailActivity.java:907)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.access$100(cgeo.geocaching.CacheDetailActivity) (CacheDetailActivity.java:186)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.CacheDetailActivity$LoadCacheHandler.handleRegularMessage(android.os.Message) (CacheDetailActivity.java:858)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.utils.DisposableHandler.handleMessage(android.os.Message) (DisposableHandler.java:61)
04-16 09:41:10.654  7557  7557 I zygote64:   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
04-16 09:41:10.654  7557  7557 I zygote64:   at void android.os.Looper.loop() (Looper.java:164)
04-16 09:41:10.654  7557  7557 I zygote64:   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6944)
04-16 09:41:10.654  7557  7557 I zygote64:   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
04-16 09:41:10.654  7557  7557 I zygote64:   at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:327)
04-16 09:41:10.654  7557  7557 I zygote64:   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1374)
04-16 09:41:10.654  7557  7557 I zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.webkit.PacProcessor" on path: DexPathList[[zip file "/data/app/com.android.chrome-uUrMuY78FGg84s2_9z9C_Q==/base.apk"],nativeLibraryDirectories=[/data/app/com.android.chrome-uUrMuY78FGg84s2_9z9C_Q==/lib/arm64, /data/app/com.android.chrome-uUrMuY78FGg84s2_9z9C_Q==/base.apk!/lib/arm64-v8a, /system/lib64, /system/vendor/lib64]]
04-16 09:41:10.654  7557  7557 I zygote64:   at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:93)
04-16 09:41:10.654  7557  7557 I zygote64:   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
04-16 09:41:10.654  7557  7557 I zygote64:   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
04-16 09:41:10.654  7557  7557 I zygote64:   at java.lang.Class java.lang.Class.classForName(java.lang.String, boolean, java.lang.ClassLoader) (Class.java:-2)
04-16 09:41:10.654  7557  7557 I zygote64:   at java.lang.Class java.lang.Class.forName(java.lang.String, boolean, java.lang.ClassLoader) (Class.java:453)
04-16 09:41:10.654  7557  7557 I zygote64:   at java.lang.Class android.webkit.WebViewFactory.getWebViewProviderClass(java.lang.ClassLoader) (WebViewFactory.java:150)
04-16 09:41:10.654  7557  7557 I zygote64:   at java.lang.Class android.webkit.WebViewFactory.getProviderClass() (WebViewFactory.java:417)
04-16 09:41:10.654  7557  7557 I zygote64:   at android.webkit.WebViewFactoryProvider android.webkit.WebViewFactory.getProvider() (WebViewFactory.java:211)
04-16 09:41:10.654  7557  7557 I zygote64:   at android.webkit.WebViewFactoryProvider android.webkit.WebView.getFactory() (WebView.java:2467)
04-16 09:41:10.654  7557  7557 I zygote64:   at java.lang.String android.webkit.WebView.findAddress(java.lang.String) (WebView.java:1738)
04-16 09:41:10.654  7557  7557 I zygote64:   at void android.text.util.Linkify.gatherMapLinks(java.util.ArrayList, android.text.Spannable) (Linkify.java:805)
04-16 09:41:10.654  7557  7557 I zygote64:   at boolean android.text.util.Linkify.addLinks(android.text.Spannable, int) (Linkify.java:481)
04-16 09:41:10.654  7557  7557 I zygote64:   at boolean android.text.util.Linkify.addLinks(android.widget.TextView, int) (Linkify.java:539)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.setPersonalNote(android.widget.TextView, android.view.View, java.lang.String) (CacheDetailActivity.java:2887)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.access$3300(android.widget.TextView, android.view.View, java.lang.String) (CacheDetailActivity.java:186)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.CacheDetailActivity$DescriptionViewCreator.setContent() (CacheDetailActivity.java:1699)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.activity.TabbedViewPagerFragment.notifyDataSetChanged() (TabbedViewPagerFragment.java:63)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.activity.TabbedViewPagerActivity.setOrderedPages(long[]) (TabbedViewPagerActivity.java:119)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.notifyDataSetChanged() (CacheDetailActivity.java:907)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.access$100(cgeo.geocaching.CacheDetailActivity) (CacheDetailActivity.java:186)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.CacheDetailActivity$LoadCacheHandler.handleRegularMessage(android.os.Message) (CacheDetailActivity.java:858)
04-16 09:41:10.654  7557  7557 I zygote64:   at void cgeo.geocaching.utils.DisposableHandler.handleMessage(android.os.Message) (DisposableHandler.java:61)
04-16 09:41:10.654  7557  7557 I zygote64:   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
04-16 09:41:10.654  7557  7557 I zygote64:   at void android.os.Looper.loop() (Looper.java:164)
04-16 09:41:10.654  7557  7557 I zygote64:   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6944)
04-16 09:41:10.654  7557  7557 I zygote64:   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
04-16 09:41:10.654  7557  7557 I zygote64:   at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:327)
04-16 09:41:10.654  7557  7557 I zygote64:   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1374)
```

and this one:
```
04-16 09:41:22.118  7557  7557 W cgeo    : [main] UncaughtException
04-16 09:41:22.118  7557  7557 W cgeo    : java.lang.ArrayIndexOutOfBoundsException: length=9; index=-2
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.text.DynamicLayout.getBlockIndex(DynamicLayout.java:646)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.widget.Editor.drawHardwareAccelerated(Editor.java:1820)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.widget.Editor.onDraw(Editor.java:1789)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.widget.TextView.onDraw(TextView.java:7763)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.draw(View.java:20373)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.updateDisplayListIfDirty(View.java:19318)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.draw(View.java:20096)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.updateDisplayListIfDirty(View.java:19309)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.draw(View.java:20096)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.draw(View.java:20376)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at com.google.android.material.textfield.TextInputLayout.draw(TextInputLayout.java:4092)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.updateDisplayListIfDirty(View.java:19318)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.draw(View.java:20096)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.updateDisplayListIfDirty(View.java:19309)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.draw(View.java:20096)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.updateDisplayListIfDirty(View.java:19309)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.draw(View.java:20096)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.updateDisplayListIfDirty(View.java:19309)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.draw(View.java:20096)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.draw(View.java:20376)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at com.android.internal.policy.DecorView.draw(DecorView.java:980)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.View.updateDisplayListIfDirty(View.java:19318)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:686)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:692)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:800)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewRootImpl.draw(ViewRootImpl.java:3501)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:3288)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2823)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1785)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7832)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.Choreographer.doCallbacks(Choreographer.java:723)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.Choreographer.doFrame(Choreographer.java:658)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.os.Handler.handleCallback(Handler.java:789)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.os.Handler.dispatchMessage(Handler.java:98)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.os.Looper.loop(Looper.java:164)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at android.app.ActivityThread.main(ActivityThread.java:6944)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at java.lang.reflect.Method.invoke(Native Method)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
04-16 09:41:22.118  7557  7557 W cgeo    : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
```

Any ideas?
Full logfile available upon request.

Addition info: Ticket number 145556

They are using a rather old version of c:geo (2022.09.04), so I asked them to upgrade and then check whether the error persists.

Another user on support mail with a similar crash (support ticket 116169):

The user claims frequent crashes when re-accessing offline logs to compose them for sending. In some sessions it happens each time, at other sessions the problem does not show up at all.

Crash found frequently in users log:
```
04-30 16:49:34.804 32630 32630 E AndroidRuntime: FATAL EXCEPTION: main
04-30 16:49:34.804 32630 32630 E AndroidRuntime: Process: cgeo.geocaching, PID: 32630
04-30 16:49:34.804 32630 32630 E AndroidRuntime: java.lang.ArrayIndexOutOfBoundsException: length=9; index=-1
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.text.DynamicLayout.getBlockIndex(DynamicLayout.java:646)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.widget.Editor.drawHardwareAccelerated(Editor.java:1820)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.widget.Editor.onDraw(Editor.java:1789)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.widget.TextView.onDraw(TextView.java:7779)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20373)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19318)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4405)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4385)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19277)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20096)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20376)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at com.google.android.material.textfield.TextInputLayout.draw(TextInputLayout.java:4092)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19318)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20096)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19309)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20096)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19309)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20096)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19309)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20096)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20376)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.widget.ScrollView.draw(ScrollView.java:2731)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19318)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20096)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19309)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20096)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19309)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20096)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19309)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20096)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19309)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20096)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.draw(View.java:20376)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at com.android.internal.policy.DecorView.draw(DecorView.java:980)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19318)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:686)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:692)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:800)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewRootImpl.draw(ViewRootImpl.java:3496)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:3283)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2818)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1780)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7827)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:723)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:658)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:789)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:98)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6944)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
04-30 16:49:34.804 32630 32630 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
```

The log also contains a similar error as mentioned in the initial post:
```
04-30 16:49:57.175   393   393 I zygote64: Rejecting re-init on previously-failed class java.lang.Class<x50>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/webkit/PacProcessor;
04-30 16:49:57.175   393   393 I zygote64:   at java.lang.Class java.lang.Class.classForName(java.lang.String, boolean, java.lang.ClassLoader) (Class.java:-2)
04-30 16:49:57.175   393   393 I zygote64:   at java.lang.Class java.lang.Class.forName(java.lang.String, boolean, java.lang.ClassLoader) (Class.java:453)
04-30 16:49:57.175   393   393 I zygote64:   at java.lang.Class android.webkit.WebViewFactory.getWebViewProviderClass(java.lang.ClassLoader) (WebViewFactory.java:150)
04-30 16:49:57.175   393   393 I zygote64:   at java.lang.Class android.webkit.WebViewFactory.getProviderClass() (WebViewFactory.java:417)
04-30 16:49:57.175   393   393 I zygote64:   at android.webkit.WebViewFactoryProvider android.webkit.WebViewFactory.getProvider() (WebViewFactory.java:211)
04-30 16:49:57.175   393   393 I zygote64:   at android.webkit.WebViewFactoryProvider android.webkit.WebView.getFactory() (WebView.java:2467)
04-30 16:49:57.175   393   393 I zygote64:   at java.lang.String android.webkit.WebView.findAddress(java.lang.String) (WebView.java:1738)
04-30 16:49:57.176   393   393 I zygote64:   at void android.text.util.Linkify.gatherMapLinks(java.util.ArrayList, android.text.Spannable) (Linkify.java:805)
04-30 16:49:57.176   393   393 I zygote64:   at boolean android.text.util.Linkify.addLinks(android.text.Spannable, int) (Linkify.java:481)
04-30 16:49:57.176   393   393 I zygote64:   at boolean android.text.util.Linkify.addLinks(android.widget.TextView, int) (Linkify.java:539)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.setPersonalNote(android.widget.TextView, android.view.View, java.lang.String) (CacheDetailActivity.java:3055)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.access$3300(android.widget.TextView, android.view.View, java.lang.String) (CacheDetailActivity.java:196)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.CacheDetailActivity$DescriptionViewCreator.setContent() (CacheDetailActivity.java:1725)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.activity.TabbedViewPagerFragment.notifyDataSetChanged() (TabbedViewPagerFragment.java:63)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.activity.TabbedViewPagerActivity.setOrderedPages(long[]) (TabbedViewPagerActivity.java:119)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.notifyDataSetChanged() (CacheDetailActivity.java:920)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.access$100(cgeo.geocaching.CacheDetailActivity) (CacheDetailActivity.java:196)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.CacheDetailActivity$LoadCacheHandler.handleRegularMessage(android.os.Message) (CacheDetailActivity.java:871)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.utils.DisposableHandler.handleMessage(android.os.Message) (DisposableHandler.java:61)
04-30 16:49:57.176   393   393 I zygote64:   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
04-30 16:49:57.176   393   393 I zygote64:   at void android.os.Looper.loop() (Looper.java:164)
04-30 16:49:57.176   393   393 I zygote64:   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6944)
04-30 16:49:57.176   393   393 I zygote64:   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
04-30 16:49:57.176   393   393 I zygote64:   at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:327)
04-30 16:49:57.176   393   393 I zygote64:   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1374)
04-30 16:49:57.176   393   393 I zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.webkit.PacProcessor" on path: DexPathList[[zip file "/data/app/com.google.android.webview-khAShu42UO0kdCLx5JC0MQ==/base.apk"],nativeLibraryDirectories=[/data/app/com.google.android.webview-khAShu42UO0kdCLx5JC0MQ==/lib/arm64, /data/app/com.google.android.webview-khAShu42UO0kdCLx5JC0MQ==/base.apk!/lib/arm64-v8a, /system/lib64, /system/vendor/lib64]]
04-30 16:49:57.176   393   393 I zygote64:   at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:93)
04-30 16:49:57.176   393   393 I zygote64:   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
04-30 16:49:57.176   393   393 I zygote64:   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
04-30 16:49:57.176   393   393 I zygote64:   at java.lang.Class java.lang.Class.classForName(java.lang.String, boolean, java.lang.ClassLoader) (Class.java:-2)
04-30 16:49:57.176   393   393 I zygote64:   at java.lang.Class java.lang.Class.forName(java.lang.String, boolean, java.lang.ClassLoader) (Class.java:453)
04-30 16:49:57.176   393   393 I zygote64:   at java.lang.Class android.webkit.WebViewFactory.getWebViewProviderClass(java.lang.ClassLoader) (WebViewFactory.java:150)
04-30 16:49:57.176   393   393 I zygote64:   at java.lang.Class android.webkit.WebViewFactory.getProviderClass() (WebViewFactory.java:417)
04-30 16:49:57.176   393   393 I zygote64:   at android.webkit.WebViewFactoryProvider android.webkit.WebViewFactory.getProvider() (WebViewFactory.java:211)
04-30 16:49:57.176   393   393 I zygote64:   at android.webkit.WebViewFactoryProvider android.webkit.WebView.getFactory() (WebView.java:2467)
04-30 16:49:57.176   393   393 I zygote64:   at java.lang.String android.webkit.WebView.findAddress(java.lang.String) (WebView.java:1738)
04-30 16:49:57.176   393   393 I zygote64:   at void android.text.util.Linkify.gatherMapLinks(java.util.ArrayList, android.text.Spannable) (Linkify.java:805)
04-30 16:49:57.176   393   393 I zygote64:   at boolean android.text.util.Linkify.addLinks(android.text.Spannable, int) (Linkify.java:481)
04-30 16:49:57.176   393   393 I zygote64:   at boolean android.text.util.Linkify.addLinks(android.widget.TextView, int) (Linkify.java:539)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.setPersonalNote(android.widget.TextView, android.view.View, java.lang.String) (CacheDetailActivity.java:3055)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.access$3300(android.widget.TextView, android.view.View, java.lang.String) (CacheDetailActivity.java:196)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.CacheDetailActivity$DescriptionViewCreator.setContent() (CacheDetailActivity.java:1725)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.activity.TabbedViewPagerFragment.notifyDataSetChanged() (TabbedViewPagerFragment.java:63)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.activity.TabbedViewPagerActivity.setOrderedPages(long[]) (TabbedViewPagerActivity.java:119)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.notifyDataSetChanged() (CacheDetailActivity.java:920)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.CacheDetailActivity.access$100(cgeo.geocaching.CacheDetailActivity) (CacheDetailActivity.java:196)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.CacheDetailActivity$LoadCacheHandler.handleRegularMessage(android.os.Message) (CacheDetailActivity.java:871)
04-30 16:49:57.176   393   393 I zygote64:   at void cgeo.geocaching.utils.DisposableHandler.handleMessage(android.os.Message) (DisposableHandler.java:61)
04-30 16:49:57.176   393   393 I zygote64:   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
04-30 16:49:57.176   393   393 I zygote64:   at void android.os.Looper.loop() (Looper.java:164)
04-30 16:49:57.176   393   393 I zygote64:   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6944)
04-30 16:49:57.176   393   393 I zygote64:   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
04-30 16:49:57.176   393   393 I zygote64:   at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:327)
04-30 16:49:57.176   393   393 I zygote64:   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1374)
```

--- a/main/src/main/java/cgeo/geocaching/CacheDetailActivity.java
+++ b/main/src/main/java/cgeo/geocaching/CacheDetailActivity.java
@@ -79,6 +79,7 @@ import cgeo.geocaching.ui.IndexOutOfBoundsAvoidingTextView;
 import cgeo.geocaching.ui.TextParam;
 import cgeo.geocaching.ui.TrackableListAdapter;
 import cgeo.geocaching.ui.UserClickListener;
+import cgeo.geocaching.ui.ViewUtils;
 import cgeo.geocaching.ui.WeakReferenceHandler;
 import cgeo.geocaching.ui.dialog.Dialogs;
 import cgeo.geocaching.ui.dialog.EditNoteDialog;
@@ -3001,7 +3002,7 @@ public class CacheDetailActivity extends TabbedViewPagerActivity
         if (StringUtils.isNotBlank(personalNote)) {
             personalNoteView.setVisibility(View.VISIBLE);
             separator.setVisibility(View.VISIBLE);
-            Linkify.addLinks(personalNoteView, Linkify.MAP_ADDRESSES | Linkify.WEB_URLS);
+            ViewUtils.safeAddLinks(personalNoteView, Linkify.MAP_ADDRESSES | Linkify.WEB_URLS);
         } else {
             personalNoteView.setVisibility(View.GONE);
             separator.setVisibility(View.GONE);

--- a/main/src/main/java/cgeo/geocaching/maps/mapsforge/v6/NewMap.java
+++ b/main/src/main/java/cgeo/geocaching/maps/mapsforge/v6/NewMap.java
@@ -1637,7 +1637,7 @@ public class NewMap extends AbstractBottomNavigationActivity implements Observer
         CharSequence message = HtmlCompat.fromHtml(attribution, HtmlCompat.FROM_HTML_MODE_LEGACY);
         if (linkify) {
             final SpannableString s = new SpannableString(message);
-            Linkify.addLinks(s, Linkify.ALL);
+            ViewUtils.safeAddLinks(s, Linkify.ALL);
             message = s;
         }
 

--- a/main/src/main/java/cgeo/geocaching/ui/TextParam.java
+++ b/main/src/main/java/cgeo/geocaching/ui/TextParam.java
@@ -9,7 +9,6 @@ import android.graphics.drawable.Drawable;
 import android.text.SpannableString;
 import android.text.TextUtils;
 import android.text.method.LinkMovementMethod;
-import android.text.util.Linkify;
 import android.widget.TextView;
 
 import androidx.annotation.NonNull;
@@ -195,7 +194,7 @@ public class TextParam {
         //linkify
         if (this.linkifyMask != 0) {
             final SpannableString linkifyString = SpannableString.valueOf(text);
-            Linkify.addLinks(linkifyString, this.linkifyMask);
+            ViewUtils.safeAddLinks(linkifyString, this.linkifyMask);
             text = linkifyString;
         }
 

--- a/main/src/main/java/cgeo/geocaching/ui/ViewUtils.java
+++ b/main/src/main/java/cgeo/geocaching/ui/ViewUtils.java
@@ -6,6 +6,7 @@ import cgeo.geocaching.databinding.CheckboxItemBinding;
 import cgeo.geocaching.databinding.DialogEdittextBinding;
 import cgeo.geocaching.ui.dialog.Dialogs;
 import cgeo.geocaching.ui.dialog.SimpleDialog;
+import cgeo.geocaching.utils.Log;
 import cgeo.geocaching.utils.functions.Action1;
 import cgeo.geocaching.utils.functions.Func1;
 import cgeo.geocaching.utils.functions.Func2;
@@ -25,6 +26,7 @@ import android.text.Selection;
 import android.text.Spannable;
 import android.text.TextWatcher;
 import android.text.method.ScrollingMovementMethod;
+import android.text.util.Linkify;
 import android.util.Pair;
 import android.view.ContextThemeWrapper;
 import android.view.Gravity;
@@ -48,6 +50,7 @@ import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.annotation.StyleRes;
 import androidx.appcompat.widget.TooltipCompat;
+import androidx.core.text.util.LinkifyCompat;
 import androidx.core.util.Consumer;
 import androidx.core.util.Predicate;
 
@@ -541,6 +544,26 @@ public class ViewUtils {
     public static void applyToView(@Nullable final View view, final Action1<View> applyMethod) {
         if (view != null) {
             applyMethod.call(view);
+        }
+    }
+
+    /** safe Linkify.addLinks version, prevents crash on older Android versions, see #14202 */
+    public static boolean safeAddLinks(@NonNull final TextView textView, @LinkifyCompat.LinkifyMask final int mask) {
+        try {
+            return Linkify.addLinks(textView, mask);
+        } catch (RuntimeException re) {
+            Log.d("Caught Error on Linkify.addLinks", re);
+            return false;
+        }
+    }
+
+    /** safe Linkify.addLinks version, prevents crash on older Android versions, see #14202 */
+    public static boolean safeAddLinks(@NonNull final Spannable sp, @LinkifyCompat.LinkifyMask final int mask) {
+        try {
+            return Linkify.addLinks(sp, mask);
+        } catch (RuntimeException re) {
+            Log.d("Caught Error on Linkify.addLinks", re);
+            return false;
         }
     }
 }

